src/infra/dex/okx/mod.rs

```diff
@@ -167,27 +183,80 @@ impl Okx {
     /// Returns a tuple of the /swap API response and dex contract address for
     /// the sell token obtained from /approve-transaction API endpoint or an
     /// error.
+    ///
+    /// Routes to sell_orders_endpoint for sell orders (exactIn) and
+    /// buy_orders_endpoint for buy orders (exactOut) if configured.
     async fn handle_api_requests(
         &self,
         order: &dex::Order,
         slippage: &dex::Slippage,
     ) -> Result<(dto::SwapResponse, eth::ContractAddress), Error> {
         let swap_request_future = async {
-            let swap_request = self.defaults.clone().try_with_domain(order, slippage)?;
-            self.send_get_request("swap", &swap_request).await
+            match order.side {
+                order::Side::Sell => {
+                    // Use V6 API for sell orders
+                    let swap_request = self.defaults.clone().with_domain(order, slippage);
+                    self.send_get_request(&self.sell_orders_endpoint, "swap", &swap_request)
+                        .await
+                }
+                order::Side::Buy => {
+                    // Use V5 API for buy orders if configured
+                    let endpoint = self
+                        .buy_orders_endpoint
+                        .as_ref()
+                        .ok_or(Error::OrderNotSupported)?;
+                    let swap_request_v6 = self.defaults.clone().with_domain(order, slippage);
+                    let swap_request_v5 = dto::SwapRequestV5::from_v6(&swap_request_v6);
+
+                    self.send_get_request(endpoint, "swap", &swap_request_v5)
+                        .await
+                }
+            }
         };
 
         let approve_transaction_request_future = async {
-            let approve_transaction_request =
-                dto::ApproveTransactionRequest::with_domain(self.defaults.chain_index, order);
-
-            let approve_transaction: dto::ApproveTransactionResponse = self
-                .send_get_request("approve-transaction", &approve_transaction_request)
-                .await?;
-
-            Ok(eth::ContractAddress(
-                approve_transaction.dex_contract_address,
-            ))
+            match order.side {
+                order::Side::Sell => {
+                    // Use V6 API for sell orders
+                    let approve_request = dto::ApproveTransactionRequest::with_domain(
```

Comment:
This is in both branches, you could pull it above the match statement so it's not duplicated.